include/gridtools/common/defs.hpp

```diff
@@ -52,8 +52,9 @@ namespace gridtools {
 #define GT_NVCC_DIAG_POP_SUPPRESS(x)
 #endif
 
-#if defined(__NVCC__) && (__CUDACC_VER_MAJOR__ == 12 && __CUDACC_VER_MINOR__ >= 1 && __CUDACC_VER_MINOR__ <= 3)
-// enables workaround for CTAD/constexpr issues in CUDA 12.1, 12.2, 12.3 (https://github.com/GridTools/gridtools/issues/1766)
+#if defined(__NVCC__) && (__CUDACC_VER_MAJOR__ == 12 && __CUDACC_VER_MINOR__ >= 1 && __CUDACC_VER_MINOR__ <= 4)
+// enables workaround for CTAD/constexpr issues in CUDA 12.1, 12.2, 12.3, 12.4
+// (https://github.com/GridTools/gridtools/issues/1766)
 #define GT_NVCC_WORKAROUND_1766 1
```

Comment:
parts of this workaround are not needed in cuda 12.4, but then there are more issues that were not in cuda 12.3